omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/VisitController1_9Test.java

```diff
@@ -214,4 +214,14 @@ public void searchByPatient_shouldGetUnretiredVisitsForThePatient() throws Excep
 		Assert.assertEquals(3, Util.getResultsSize(result));
 	}
 	
+	@Test
+	public void searchByPatient_shouldGetAllVisitsForThePatientIncludingRetired() throws Exception {
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("reason", "void test reason")));
+		SimpleObject resultWithVoidedVisits = deserialize(handle(newGetRequest(getURI(), new Parameter("patient",
+		        "da7f524f-27ce-4bb2-86d6-6d1d05312bd5"), new Parameter("includeAll", "true"))));
+		SimpleObject resultVithoutVoidedVisits = deserialize(handle(newGetRequest(getURI(), new Parameter("patient",
+		        "da7f524f-27ce-4bb2-86d6-6d1d05312bd5"))));
+		Assert.assertNotEquals(Util.getResultsSize(resultWithVoidedVisits), Util.getResultsSize(resultVithoutVoidedVisits));
```

Comment:
Instead of assertNotEquals can you assert the expected results for `resultWithVoidedVisits` and `resultVithoutVoidedVisits`

Reply:
@adamsdenniskariuki I do not understand what you mean, can you make it clearer? The point is that resultWithVoidedVisits and resultVithoutVoidedVisits can not have the same size since we are sure we voided one of the visits.

Reply:
Would it be clearer to use assertTrue() on `resultWithVoidedVisits` and `resultVithoutVoidedVisits` with their correct values.

Reply:
Ok, I tried explaining to you what I am testing. You cannot use assert true when you are expecting size. If you are expecting value of 13, would you use assertTrue? The right thing to use is assertEquals. So in this case, I am expecting the size of getVisits with voided visits not to be the same size of ones without voided. Does that makes sense?

Reply:
I meant to write `assertEquals()`. Thats correct.

Reply:
Ok, but I am testing for inequality thats why I chose assertNotEquals. But depending on convention, I know I can use assertTrue(resultWithVoidedVisits > resultVithoutVoidedVisits)

Reply:
I meant creating 2 assertEquals and checking the values for each.

Reply:
Ok, in this case, I am not sure of the exact values. I am not sure if it would be 4,2,5 etc. Leading to th

Reply:
Ok. The original is correct.

Reply:
Thank you.